sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -31,64 +31,473 @@
+
+import sdc
 
 from numba import types
 from numba.extending import (overload, overload_method, overload_attribute)
+from sdc.hiframes.pd_dataframe_ext import DataFrameType
 from numba.errors import TypingError
+from sdc.datatypes.hpat_pandas_series_functions import TypeChecker
+
+if not sdc.config.use_default_dataframe:
+    from sdc.datatypes.hpat_pandas_dataframe_types import DataFrameType
+
+    @overload_method(DataFrameType, 'count')
+    def sdc_pandas_dataframe_count(self, axis=0, level=None, numeric_only=False):
+        """
+        Pandas DataFrame method :meth:`pandas.DataFrame.count` implementation.
+
+        .. only:: developer
+
+            Test: python -m sdc.runtests sdc.tests.test_dataframe.TestDataFrame.test_count
+
+        Parameters
+        -----------
+        self: :class:`pandas.DataFrame`
+            input arg
+        axis:
+            *unsupported*
+        level:
+            *unsupported*
+        numeric_only:
+            *unsupported*
+
+        Returns
+        -------
+        :obj:`pandas.Series` or `pandas.DataFrame`
+                returns: For each column/row the number of non-NA/null entries. If level is specified returns
+                a DataFrame.
+        """
+
+        _func_name = 'Method pandas.dataframe.count().'
+
+        if not isinstance(self, DataFrameType):
+            raise TypingError('{} The object must be a pandas.dataframe. Given: {}'.format(_func_name, self))
+
+        if not (isinstance(axis, types.Omitted) or axis == 0):
+            raise TypingError("{} 'axis' unsupported. Given: {}".format(_func_name, axis))
+
+        if not (isinstance(level, types.Omitted) or level is None):
+            raise TypingError("{} 'level' unsupported. Given: {}".format(_func_name, axis))
+
+        if not (isinstance(numeric_only, types.Omitted) or numeric_only is False):
+            raise TypingError("{} 'numeric_only' unsupported. Given: {}".format(_func_name, axis))
+
+        def sdc_pandas_dataframe_count_impl(self, axis=0, level=None, numeric_only=False):
+            result_data = []
+            result_index = []
+
+            for dataframe_item in self._data:
+                item_count = dataframe_item.count()
+                item_name = dataframe_item._name
+                result_data.append(item_count)
+                result_index.append(item_name)
+
+            return pandas.Series(data=result_data, index=result_index)
+
+        return sdc_pandas_dataframe_count_impl
+
+else:
+    def sdc_pandas_dataframe_reduce_columns(df, name, param):
+        saved_columns = df.columns
+        n_cols = len(saved_columns)
+        data_args = tuple('data{}'.format(i) for i in range(n_cols))
+        help_param = ', {}={}):'
+        func_text = 'def _reduce_impl(df):'
+        for key, value in param:
+            func_text = func_text[:-2]
+            func_text = func_text + help_param
+            func_text = func_text.format(key, value)
+        func_lines = [func_text]
```

Comment:
Let's refactor this part:
```
all_params = ['df']
for name, value in param:
    all_params.append('{}={}'.format(name, value))
func_definition =  'def _reduce_impl({}):'.format(', '.join(all_params))
func_lines = [func_definition]
```

Reply:
I don't know why, but this variant not work. An error occurs. Parametrs become Series.